deps/exokit-bindings/webglcontext/src/webgl.cc

```diff
@@ -4847,6 +4858,77 @@ NAN_METHOD(WebGL2RenderingContext::New) {
   info.GetReturnValue().Set(gl2Obj);
 }
 
+// reference used https://www.khronos.org/registry/OpenGL-Refpages/es3.0/
+NAN_METHOD(WebGL2RenderingContext::CreateQuery) { // adapted from CreateBuffer
+  GLuint queryId;
+  glGenQueries(1, &queryId);
+
+  Local<Object> queryObject = Nan::New<Object>();
+  queryObject->Set(JS_STR("id"), JS_INT(queryId));
+  info.GetReturnValue().Set(queryObject);
+}
+
+NAN_METHOD(WebGL2RenderingContext::BeginQuery) { // adapted from BindBuffer
+  GLenum target = info[0]->Int32Value();
+  GLuint query = info[1]->IsObject() ? info[1]->ToObject()->Get(JS_STR("id"))->Uint32Value() : 0;
+
+  glBeginQuery(target, query);
+}
+
+NAN_METHOD(WebGL2RenderingContext::EndQuery) {
+  GLenum target = info[0]->Int32Value();
+  glEndQuery(target);
+}
+
+NAN_METHOD(WebGL2RenderingContext::GetQuery) {
+  GLenum target = info[0]->Int32Value();
+  GLenum pname = info[1]->Int32Value();
+  GLint value;
+  
+  glGetQueryiv(target, pname, &value);
+  Local<Object> queryObject = Nan::New<Object>();
+  queryObject->Set(JS_STR("id"), JS_INT(value));
+
+  info.GetReturnValue().Set(queryObject);
+}
+
+NAN_METHOD(WebGL2RenderingContext::GetQueryParameter) { // adapted from GetProgramParameter
+  GLint queryId = info[0]->ToObject()->Get(JS_STR("id"))->Int32Value();
+  GLenum pname = info[1]->Int32Value();
+  
+  switch (pname) {
+    case GL_QUERY_RESULT_AVAILABLE:
```

Comment:
I think these constants also need to be exposed on the context: https://developer.mozilla.org/en-US/docs/Web/API/WebGL2RenderingContext/getQueryParameter#Parameters

Reply:
The constants are all exposed here: https://github.com/webmixedreality/exokit/blob/6c9c7915cdf85bd99da8b16be775af06ebddf23c/deps/exokit-bindings/webglcontext/src/webgl.cc#L73

Reply:
We should make sure to get them all for the other functions as well, such as `beginQuery`: https://developer.mozilla.org/en-US/docs/Web/API/WebGL2RenderingContext/beginQuery#Parameters